### PR TITLE
Portafly/portafly uxspecs modal

### PR DESCRIPTION
**Pull request by cntlsn**
**What this PR does / why we need it**:

Adds specs to `modal.md` to cover the following use cases:
* user dismisses modal after entering text
* message sent success
* message sent error

**Which issue(s) this PR fixes** 

None

**Verification steps** 

1. Read the file

**Special notes for your reviewer**:

None

### CODE REVIEW

File: portafly/doc/ux_specs/global_components/modal.md [hunk 1]
```diff
@@ -26,6 +27,18 @@
...
+      * spinner should stay visible for at least 1.5 seconds (even when the request is resolved earlier)
+    * and the input fields are disabled
+  * If message is sent successfully:
+    * the modal will auto-dismiss
+    * and a success toast alert will be displayed
+  * If request returns an error:
+    * the modal won't auto dismiss
+    * the spinner will be replaced by the submit button
+    * the input fields will be enabled again (to allow user eventually copy paste their content)
+    * an inline error message will be displayed above the form buttons
```
Comment (josemigallas) [inline portafly/doc/ux_specs/global_components/modal.md:41]: ?
```suggestion
    * an inline warning message will be displayed above the form buttons
```

File: portafly/doc/ux_specs/global_components/modal.md [hunk 2]
```diff
@@ -7,9 +7,10 @@
 
 ### Global modal specs
 * Modal dialogs are placed in the exact centre of the page.
-* Modal dialogs feature a "X" icon in the top right corner. Clicking the icon will dismiss the modal.
+* Modal dialogs feature an "X" icon in the top right corner. Clicking the icon will dismiss the modal.
 * Clicking on the foreground (everywhere on the page outside of the area covered by the modal), will dismiss the modal.
```
Comment (josemigallas) [inline portafly/doc/ux_specs/global_components/modal.md:11]: This isn't supported by PF React. It should? https://www.patternfly.org/v4/documentation/react/components/modal

File: portafly/doc/ux_specs/global_components/modal.md [hunk 3]
```diff
@@ -26,6 +27,18 @@
     * Submit button [Disabled state until all conditions required to submit the form are met]
     * Cancel button [Always enabled]
       * Dismisses modal
+  * When user submits the form:
+    * the Submit button is replaced by a spinner
+      * spinner should stay visible for at least 1.5 seconds (even when the request is resolved earlier)
```
Comment (josemigallas) [inline portafly/doc/ux_specs/global_components/modal.md:32]: I would suggest deferring this for later, when we implement the API requests properly.
@didierofrivia

Comment (cntlsn) [reply portafly/doc/ux_specs/global_components/modal.md:32]: sounds good to me, but how to make sure we don't forget about this?

Comment (didierofrivia) [reply portafly/doc/ux_specs/global_components/modal.md:32]: Maybe Jira issue?

**Pull request merged**